source/scale/high-availability-cluster-based-deployment.rst

```diff
@@ -8,462 +8,414 @@ High availability cluster-based deployment
+For advanced settings, see the :ref:`high availability configuration settings <configure/environment-configuration-settings:high availability>` documentation.
 
-1. Go to **System Console > Environment > Web Server**, then select **Reload Configuration from Disk** to reload configuration settings for the Mattermost server from ``config.json``.
-2. Go to **System Console > Environment > Database**, then select **Recycle Database Connections** to take down existing database connections and set up new connections in the multi-database configuration.
+File descriptors and process limits
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-While the connection settings are changing, there might be a brief moment when writes to the master database are unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening will have an experience similar to losing connection to the Mattermost server.
+Update file/process limits in ``/etc/security/limits.conf``:
 
-Manual failover for master database
-```````````````````````````````````
+- ``soft nofile 65536``
+- ``hard nofile 65536``
+- ``soft nproc 8192``
+- ``hard nproc 8192``
 
-If the need arises to switch from the current master database - for example, if it is running out of disk space, or requires maintenance updates, or for other reasons - you can switch Mattermost server to use one of its read replicas as a master database by updating ``DataSource`` in ``config.json``.
+Network settings
+^^^^^^^^^^^^^^^^
 
-To apply the settings without shutting down the Mattermost server:
+Optimize network configurations in ``/etc/sysctl.conf`` on each machine that hosts a Mattermost server:
 
-1. Go to **System Console > Environment > Web Server**, then select **Reload Configuration from Disk** to reload configuration settings for the Mattermost server from ``config.json``.
-2. Go to **System Console > Environment > Database**, then select **Recycle Database Connections** to take down existing database connections and set up new connections in the multi-database configuration.
+.. code-block:: bash
 
-While the connection settings are changing, there might be a brief moment when writes to the master database are unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening can have an experience similar to losing connection to the Mattermost server.
+  net.ipv4.ip_local_port_range = 1025 65000
+  net.ipv4.tcp_fin_timeout = 30
+  net.ipv4.tcp_tw_reuse = 1
+  net.core.somaxconn = 4096
+  net.ipv4.tcp_max_syn_backlog = 8192
+  vm.min_free_kbytes = 167772
+  net.ipv4.tcp_congestion_control = bbr
+  net.core.default_qdisc = fq
+  net.ipv4.tcp_mem = 1638400 1638400 1638400
```

Comment:
Why did we remove the comments in the code block in the old document? They add useful context. There are also settings missing, like `net.ipv4.tcp_slow_start_after_idle = 0`, `net.ipv4.tcp_notsent_lowat = 16384`, and this block altogether:
```
net.ipv4.tcp_rmem = 4096 156250 625000
net.ipv4.tcp_wmem = 4096 156250 625000
net.core.rmem_max = 312500
net.core.wmem_max = 312500
net.core.rmem_default = 312500
net.core.wmem_default = 312500
```